api/controllers/site.js

```diff
@@ -95,6 +99,37 @@ module.exports = {
       });
   },
 
+  removeUser: (req, res) => {
+    const { site_id: siteId, user_id: userId } = req.params;
+
+    if (!Number(siteId) || !Number(userId)) {
+      return res.error(400);
+    }
+
+    return Site.withUsers(siteId).then((site) => {
+      if (!site) {
+        throw 404;
+      }
+
+      if (site.Users.length === 1) {
+        throw {
+          status: 400,
+          message: 'A site must have at least one user',
+        };
+      }
+
+      return authorizer.destroy(req.user, site).then(() => site);
```

Comment:
This authorizer is for checking whether a user can delete a site. We may want to add a new action to the authorizer for removing users.

Reply:
I think the only stipulation to a user being able to delete a site is that they have access to the site (and have write permissions on GitHub). Would the new authorizer action just check for both of those?  I guess I'm saying that the check the authorizer is performing still applies in this case.

Reply:
Yeah, even if the new action just calls the old action it works. I think what we'd want to avoid is a situation where we change the logic we use to check whether a user can delete a site, and inadvertently change the logic we use to check whether a user can delete another user from a site. Adding a new action to the authorizer makes it explicit that these are 2 distinct activities.

Reply:
Ok, since the authorizer also overloads the `create` action for adding a new user, we should create a separate method to handle that as well, correct?

Reply:
Yeah, I think so